gammagl/layers/conv/compgcn_conv.py

```diff
@@ -0,0 +1,195 @@
+        x = self.update(x)
+        return x
+
+    def __collect__(self, x, edge_index,edge_type, aggr, kwargs):
+        out = {}
+
+        for k, v in kwargs.items():
+            out[k] = v
+        out['x'] = x
+        out['edge_index'] = edge_index
+        out['aggr'] = aggr
+        out['edge_type'] = edge_type
+        return out
+
+    def message(self, x, edge_index,edge_type, edge_weight=None,rel_emb=None,linear=None):
+        """
+        Function that construct message from source nodes to destination nodes.
+        Args:
+            x (Tensor): input node feature
+            edge_index (Tensor): edges from src to dst
+            edge_weight (Tensor): weight of each edge
+        Returns:
+
+        """
+        import torch
+       # rel_emb = rel_emb[edge_type]
+        rel_emb = torch.index_select(rel_emb, 0 ,edge_type)
+        x_emb = x[edge_index[1]]
+        if self.op == 'sub': x_rel_emb = x_emb - rel_emb
+        elif self.op == 'mult': x_rel_emb = x_emb * rel_emb
+        elif self.op == 'corr': x_rel_emb = self.ccorr(x_emb,rel_emb)
+        msg = linear(x_rel_emb)
+        if edge_weight is not None:
+            edge_weight = tlx.expand_dims(edge_weight, -1)
+            return msg * edge_weight
+        else:
+            return msg
+
+    def aggregate(self, msg, edge_index, num_nodes=None, aggr='sum',dim_size=None):
+        """
+        Function that aggregates message from edges to destination nodes.
+        Args:
+            msg (Tensor): message construct by message function
+            edge_index (Tensor): edges from src to dst
+            num_nodes (int): number of nodes of the graph
+            aggr (str): aggregation type, default = 'sum', optional=['sum', 'mean', 'max']
+
+        Returns:
+
+        """
+        dst_index = edge_index[0, :]
+        if aggr == 'sum':
+            return unsorted_segment_sum(msg, dst_index, num_nodes)
+            #return unsorted_segment_sum(msg, dst_index, num_nodes)
+        elif aggr == 'mean':
+            return unsorted_segment_mean(msg, dst_index, num_nodes)
+        elif aggr == 'max':
+            return unsorted_segment_max(msg, dst_index, num_nodes)
+        else:
+            raise NotImplementedError('Not support for this opearator')
+
+
+    def ccorr(self, a, b):
+        """
+        Compute circular correlation of two tensors.
+        Parameters
+        ----------
+        a: Tensor, 1D or 2D
+        b: Tensor, 1D or 2D
+        Notes
+        -----
+        Input a and b should have the same dimensions. And this operation supports broadcasting.
+        Returns
+        -------
+        Tensor, having the same dimension as the input a.
+        """
+
+        def com_mult(a, b):
+            r1, i1 = a[..., 0], a[..., 1]
+            r2, i2 = b[..., 0], b[..., 1]
+            return torch.stack([r1 * r2 - i1 * i2, r1 * i2 + i1 * r2], dim=-1)
```

Comment:
Do not use torch in the ``conv`` layer. Change to ``tensorlayerx``.

Reply:
All these are used for fast fourier transform, but tensorlaye doesn't have it. I think we can temporarily use pytorch